### PR TITLE
[types-pillow] update deprecated `PIL.Image` constants

### LINKED ISSUE #8366: [Pillow] Module has no attribute "Resampling"
Hi,

Since Pillow version >9.1.0, the Pillow library has switched the `PIL.Image` filters to `Resampling` enum class and deprecated the old method of using typing.Literal for defining filters. Due to this reason, mypy shows the error stated in this issue title if anyone is using types-Pillow package and is using `PIL.Image.Resampling` approach in our project. The reason for me to switch to Resampling enum method is to avoid DeprecationWarning and safeguard my code ahead of time.
```py
[WARNING] py.warnings: DeprecationWarning: LANCZOS is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.LANCZOS instead.
```

Pillow docs for reference: https://pillow.readthedocs.io/en/stable/reference/Image.html?highlight=lanczos#resampling-filters
Pillow 9.1.0 release notes: https://pillow.readthedocs.io/en/stable/releasenotes/9.1.0.html#constants
the same is documented in https://pillow.readthedocs.io/en/stable/deprecations.html#constants

It would be grateful if this can be accounted for in the pillow typeshed package if possible. I would like to submit a PR to fix but unsure how should I go about this so I decided to open an issue.

### PR DESCRIPTION
**Pull request by ow0x**
Hi,

As per my opened issue #8366 , this PR updates deprecated `PIL.Image` constants (since Pillow >9.1.0) to the documented ones as per Pillow library documentation.
Pillow docs for reference: https://pillow.readthedocs.io/en/stable/deprecations.html#constants
Reference to Pillow v9.1.0 release notes where the deprecation notice was intially documented: https://pillow.readthedocs.io/en/stable/releasenotes/9.1.0.html#constants

I would like to wholeheartedly thank @hauntsaninja for providing the guidance on how and where to make necessary changes and the valuable opportunity to contribute to this open source project.

I have read the contribution guidelines of this project and tried to follow them to the best of my abilities. Please guide me on right path in case I have made the changes inappropriately as this is my first PR on this project and I am a bit unsure if I have done it right! 😅 

This PR resolves #8366

### CODE REVIEW

**Comment by srittau** [comment]
Thanks! The old numeric constants still work (with a warning), right? In that case we should keep them for now. The primer output show that a quite a few projects still use the outdated constants.

**Comment by AlexWaygood** [comment]
One of us will take a look soon and try to help you out with the primer diff :)

File: stubs/Pillow/PIL/Image.pyi [hunk 1]
```diff
@@ -204,13 +245,13 @@ class Image:
     def split(self) -> tuple[Image, ...]: ...
     def getchannel(self, channel: int | str) -> Image: ...
     def tell(self) -> int: ...
-    def thumbnail(self, size: tuple[int, int], resample: _Resample = ..., reducing_gap: float = ...) -> None: ...
+    def thumbnail(self, size: tuple[int, int], resample: Resampling | _Resample = ..., reducing_gap: float = ...) -> None: ...
     def transform(
         self,
         size: _Size,
         method: Literal[0, 1, 2, 3, 4],
```
Comment (hauntsaninja) [inline stubs/Pillow/PIL/Image.pyi:252]: Maybe out of scope for this PR, but this kind of thing will get rid of some more of the mypy_primer hits
```suggestion
        method: Transform | Literal[0, 1, 2, 3, 4],
```

But overall I wouldn't worry too much about the remaining mypy_primer hits. The ones in `_pil_constants` are unavoidable and many of the others were preexisting

**Comment by AlexWaygood** [comment]
> ok, thank you very much!

Thank _you_ for the PR!

**Review by hauntsaninja — APPROVED** [review]
Thank you for filing the issue and the fix, this lgtm!

**Comment by hauntsaninja** [comment]
There's a similar change you could make to `def transpose` by making `method` be `Transpose` as well as Literal (Github wouldn't let me add the suggestion in the code review)

**Comment by hauntsaninja** [comment]
Thank you again! :-)

**Pull request merged**